anime_downloader/downloader/base_downloader.py

```diff
@@ -28,25 +28,35 @@ def __init__(self, source, path, force, range_size, callback=None):
 
     def check_if_exists(self):
         # Added Referer Header as kwik needd it.
-        headers = {
-            'user-agent': "Mozilla/5.0 (Windows NT 10.0; Win64; x64) Gecko/20100101Firefox/56.0",
-        }
+        headers = self.source.headers
+        if 'user-agent' not in headers:
+            headers['user-agent'] = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) Gecko/20100101Firefox/56.0",
+
         if self.source.referer:
             headers['referer'] = self.source.referer
 
-        # using session downloads the whole file, essentially freezing the program.
-        r = requests.get(
-            self.source.stream_url, headers=headers, stream=True)
-
-        self._total_size = int(r.headers['Content-length'])
-        logger.debug('total size: ' + str(self._total_size))
-        if os.path.exists(self.path):
-            if abs(os.stat(self.path).st_size - self._total_size) < 10 \
-               and not self.force:
-                logger.warning('File already downloaded. Skipping download.')
+        for i in range(5):
```

Comment:
What does this do?

Reply:
Just simple retries if the first request somehow failed